Sources/OTel/Metrics/SwiftMetricsFactory/OTLPMetricsFactory.swift

```diff
@@ -0,0 +1,190 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2024 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftPrometheus open source project
+//
+// Copyright (c) 2018-2023 SwiftPrometheus project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftPrometheus project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CoreMetrics
+
+/// A Swift Metrics `MetricsFactory` implementation backed by ``OTelMetricRegistry``.
+///
+/// - TODO: Review the API surface we want to provide.
+///
+///   Right now, this is a wrapper type around the ``OTelMetricRegistry`` which adds some configuration. The API is
+///   similar to what is in Swift Prometheus, but we might not need to layer this way.
+///
+///   Specifically, Swift Prometheus provides public API on the registry itself for directly creating and exporting
+///   metrics, without using the Swift Metrics API.
+///
+///   Currently, the registry in this package is opaque, which begs the question: maybe the registry type should be
+///   internal and we just have this factory be the public API for now and defer making the registry public to if/when
+///   we want to extend this package to provide direct OTel funcionality.
+public struct OTLPMetricsFactory: Sendable {
```

Comment:
I'm a bit unsure about the name and/or location of this one. Until now, everything `OTLP` specific was either in `OTLPCore` or `OTLPGRPC`. I did that to leave room to support other export protocols if needed. I'm happy to leave it like this for now, but perhaps worth considering when we go over the public API surface in the end to check if it shouldn't live somewhere else or be named differently.

Reply:
We should discuss the objection, which I take to be that we're doing something called `OTLP*` here. I named it this to imply that we're doing the bare minimum to export Swift Metrics as OTLP (cf. being able to express the full power of OTel), but we can just as easily name this `OTelMetricsFactory`, if you like.

Now I've marked everything SPI, and added a task to bikeshed the public API to the covering issue, maybe we punt on it to keep things moving?

Reply:
Yeah, happy to move on for now and circle back later.